test/kubemark/gce/util.sh

```diff
@@ -31,127 +31,32 @@ function authenticate-docker {
-    --project "${PROJECT}" \
-    --region "${REGION}" -q
-
-    MASTER_IP=$(gcloud compute addresses describe "${MASTER_NAME}-ip" \
-    --project "${PROJECT}" --region "${REGION}" -q --format='value(address)')
-  fi
-}
-
-function create-master-instance-with-resources {
-  GCLOUD_COMMON_ARGS=(--project "${PROJECT}" --zone "${ZONE}")
-  # Override the master image project to cos-cloud for COS images staring with `cos` string prefix.
-  DEFAULT_GCI_PROJECT=google-containers
-  if [[ "${GCI_VERSION}" == "cos"* ]]; then
-      DEFAULT_GCI_PROJECT=cos-cloud
-  fi
-  MASTER_IMAGE_PROJECT=${KUBE_GCE_MASTER_PROJECT:-${DEFAULT_GCI_PROJECT}}
-
-  run-gcloud-compute-with-retries disks create "${MASTER_NAME}-pd" \
-    "${GCLOUD_COMMON_ARGS[@]}" \
-    --type "${MASTER_DISK_TYPE}" \
-    --size "${MASTER_DISK_SIZE}" &
-
-  if [ "${EVENT_PD:-}" == "true" ]; then
-    run-gcloud-compute-with-retries disks create "${MASTER_NAME}-event-pd" \
-      "${GCLOUD_COMMON_ARGS[@]}" \
-      --type "${MASTER_DISK_TYPE}" \
-      --size "${MASTER_DISK_SIZE}" &
-  fi
-
-  get-or-create-master-ip &
-
-  wait
-
-  run-gcloud-compute-with-retries instances create "${MASTER_NAME}" \
-    "${GCLOUD_COMMON_ARGS[@]}" \
-    --address "${MASTER_IP}" \
-    --machine-type "${MASTER_SIZE}" \
-    --image-project="${MASTER_IMAGE_PROJECT}" \
-    --image "${MASTER_IMAGE}" \
-    --tags "${MASTER_TAG}" \
-    --subnet "${SUBNETWORK:-${NETWORK}}" \
-    --scopes "storage-ro,logging-write" \
-    --boot-disk-size "${MASTER_ROOT_DISK_SIZE}" \
-    --disk "name=${MASTER_NAME}-pd,device-name=master-pd,mode=rw,boot=no,auto-delete=no"
-
-  run-gcloud-compute-with-retries instances add-metadata "${MASTER_NAME}" \
-    "${GCLOUD_COMMON_ARGS[@]}" \
-    --metadata-from-file startup-script="${KUBE_ROOT}/test/kubemark/resources/start-kubemark-master.sh" &
-
-  if [ "${EVENT_PD:-}" == "true" ]; then
-    echo "Attaching ${MASTER_NAME}-event-pd to ${MASTER_NAME}"
-    run-gcloud-compute-with-retries instances attach-disk "${MASTER_NAME}" \
-   "${GCLOUD_COMMON_ARGS[@]}" \
-    --disk "${MASTER_NAME}-event-pd" \
-    --device-name="master-event-pd" &
-  fi
-
-  run-gcloud-compute-with-retries firewall-rules create "${MASTER_NAME}-https" \
-    --project "${PROJECT}" \
-    --network "${NETWORK}" \
-    --source-ranges "0.0.0.0/0" \
-    --target-tags "${MASTER_TAG}" \
-    --allow "tcp:443" &
-
-  run-gcloud-compute-with-retries firewall-rules create "${MASTER_NAME}-internal" \
-    --project "${PROJECT}" \
-    --network "${NETWORK}" \
-    --source-ranges "10.0.0.0/8" \
-    --target-tags "${MASTER_TAG}" \
-    --allow "tcp:1-2379,tcp:2382-65535,udp:1-65535,icmp" &
-
-  wait
-}
-
-# Command to be executed is '$1'.
-# No. of retries is '$2' (if provided) or 1 (default).
-function execute-cmd-on-master-with-retries() {
-  RETRIES="${2:-1}" run-gcloud-compute-with-retries ssh "${MASTER_NAME}" --zone="${ZONE}" --project="${PROJECT}" --command="$1"
+function create-kubemark-master {
```

Comment:
One thing to ensure (maybe it's already done) is that kubemark was restructured so that it can be run on different providers and IIRC it's now mostly about implementing this interface:
https://github.com/kubernetes/kubernetes/tree/master/test/kubemark/skeleton

Given people are running kubemarks on other providers, we don't want to break that.

Reply:
The only logic for starting kubemark that lives in k8s repo is iks. This PR won't break it as iks/startup.sh copies all the script with 'exit 0' at the very end.

After changing the interface (IIUC) it should be to remove copy from iks/startup.sh (the new interface can be implemented for iks).